pyomo/contrib/mindtpy/util.py

```diff
@@ -844,3 +849,65 @@ def set_up_logger(config):
+    dual_bound_progress = solve_data.dual_bound_progress.copy()
+    # Initial dual bound is set to inf or -inf. To calculate dual integral, we set
+    # initial_dual_bound to 10% greater or smaller than the first_found_dual_bound.
+    # TODO: check if the calculation of initial_dual_bound needs to be modified.
+    for dual_bound in dual_bound_progress:
+        if dual_bound != dual_bound_progress[0]:
+            break
+    for i in range(len(dual_bound_progress)):
+        if dual_bound_progress[i] == solve_data.dual_bound_progress[0]:
+            if solve_data.objective_sense == minimize:
+                if dual_bound > 0:
+                    dual_bound_progress[i] = dual_bound * 0.9
+                else:
+                    dual_bound_progress[i] = dual_bound * 1.1
+            else:
+                if dual_bound > 0:
+                    dual_bound_progress[i] = dual_bound * 1.1
+                else:
+                    dual_bound_progress[i] = dual_bound * 0.9
+        else:
+            break
+    for i in range(len(dual_bound_progress)):
+        if i == 0:
+            dual_integral += abs(dual_bound_progress[i] - solve_data.dual_bound) * (solve_data.dual_bound_progress_time[i])
+        else:
+            dual_integral += abs(dual_bound_progress[i] - solve_data.dual_bound) * (solve_data.dual_bound_progress_time[i] - solve_data.dual_bound_progress_time[i-1])
+    return dual_integral
+
+
+def get_primal_integral(solve_data):
+    primal_integral = 0
+    primal_bound_progress = solve_data.primal_bound_progress.copy()
+    # Initial primal bound is set to inf or -inf. To calculate primal integral, we set
+    # initial_primal_bound to 10% greater or smaller than the first_found_primal_bound.
+    # TODO: check if the calculation of initial_primal_bound needs to be modified.
+    for primal_bound in primal_bound_progress:
+        if primal_bound != primal_bound_progress[0]:
+            break
+    for i in range(len(primal_bound_progress)):
+        if primal_bound_progress[i] == solve_data.primal_bound_progress[0]:
+            if solve_data.objective_sense == minimize:
+                if primal_bound > 0:
+                    primal_bound_progress[i] = primal_bound * 1.1
+                else:
+                    primal_bound_progress[i] = primal_bound * 0.9
+            else:
+                if primal_bound > 0:
+                    primal_bound_progress[i] = primal_bound * 0.9
+                else:
+                    primal_bound_progress[i] = primal_bound * 1.1
+        else:
+            break
+    for i in range(len(primal_bound_progress)):
+        if i == 0:
+            primal_integral += abs(primal_bound_progress[i] - solve_data.primal_bound) * (solve_data.primal_bound_progress_time[i])
+        else:
+            primal_integral += abs(primal_bound_progress[i] - solve_data.primal_bound) * (solve_data.primal_bound_progress_time[i] - solve_data.primal_bound_progress_time[i-1])
+    return primal_integral
```

Comment:
These integrals should be reported at the end of any MindtPy run, for both benchmarking and user information reasons.
Log it into info with a statement regarding the 10% extra at the initial computation.

Reply:
I have reported them in `result.solver`. Does it still need to appear in the log?

Reply:
I have added the integrals in the log.